### PR TITLE
Re-enable skipped test `Test_RecipeCommands`

### LINKED ISSUE #6776: Re-enable skipped functional test: `Test_RecipeCommands`
Just noticed that [Test_RecipeCommands](https://github.com/radius-project/radius/blob/main/test/functional/shared/cli/cli_test.go#L694) functional test was disabled as a part of migration to GHCR effort here: https://github.com/radius-project/radius/commit/a1971314d5098290e1c9b271bf5b388384ab8bfc. It's unclear what the plan to re-enable the test is, so creating this issue to ensure we get back to this and have test coverage here.

[AB#10306](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10306)

### PR DESCRIPTION
**Pull request by sk593**
# Description

This re-enables `Test_RecipeCommands` 

## Type of change

- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius #6776

Fixes: #6776

### CODE REVIEW

File: test/functional/shared/cli/cli_test.go [hunk 1]
```diff
@@ -66,31 +66,37 @@ func verifyRecipeCLI(ctx context.Context, t *testing.T, test shared.RPTest) {
...
 	envName := test.Steps[0].RPResources.Resources[0].Name
 	recipeName := "recipeName"
-	recipeTemplate := "ghcr.io/testpublicrecipe/bicep/modules/testTemplate:v1"
-	templateKind := "bicep"
-	resourceType := "Applications.Datastores/mongoDatabases"
-	file := "testdata/corerp-redis-recipe.bicep"
+	recipe1 := "recipe1"
+	recipe2 := "recipe2"
+	registry := strings.TrimPrefix(functional.GetBicepRecipeRegistry(), "registry=")
+	version := strings.TrimPrefix(functional.GetBicepRecipeVersion(), "version=")
```
Comment (ytimocin) [inline test/functional/shared/cli/cli_test.go:72]: Why do we need to do this?

Comment (sk593) [reply test/functional/shared/cli/cli_test.go:72]: These functions prefix the registry and version strings so we need to remove them for this test. https://github.com/radius-project/radius/blob/c262df7aee22cf1ebea2fa66d3e70cfdaa135ca0/test/functional/testUtil.go#L91

File: test/functional/shared/cli/cli_test.go [hunk 2]
```diff
@@ -66,31 +66,37 @@ func verifyRecipeCLI(ctx context.Context, t *testing.T, test shared.RPTest) {
 	// for AWS test and has the AWS scope which the environment created in this does not.
```
Comment (kachawla) [inline test/functional/shared/cli/cli_test.go:66]: Unrelated to this pr, but looking into this function overall - Is this comment still valid? Doesn't look the environment is getting switched anywhere in this function.

File: test/functional/shared/cli/cli_test.go [hunk 3]
```diff
@@ -66,31 +66,37 @@ func verifyRecipeCLI(ctx context.Context, t *testing.T, test shared.RPTest) {
...
-	file := "testdata/corerp-redis-recipe.bicep"
+	recipe1 := "recipe1"
+	recipe2 := "recipe2"
+	registry := strings.TrimPrefix(functional.GetBicepRecipeRegistry(), "registry=")
+	version := strings.TrimPrefix(functional.GetBicepRecipeVersion(), "version=")
+	recipeTemplate := fmt.Sprintf("%s/recipes/local-dev/rediscaches:%s", registry, version)
+	envRecipeTemplateBicep := fmt.Sprintf("%s/test/functional/shared/recipes/corerp-redis-recipe:%s", registry, version)
+	envRecipeTemplateTerraform := "Azure/cosmosdb/azurerm"
+	templateKindBicep := "bicep"
+	templateKindTerraform := "terraform"
```
Comment (kachawla) [inline test/functional/shared/cli/cli_test.go:77]: Some suggestions for readability, let me know if they don't make sense.

```suggestion
        registry := strings.TrimPrefix(functional.GetBicepRecipeRegistry(), "registry=")
	version := strings.TrimPrefix(functional.GetBicepRecipeVersion(), "version=")
	
        recipeName := "recipeName"
        recipeTemplate := fmt.Sprintf("%s/recipes/local-dev/rediscaches:%s", registry, version)
        
	bicepRecipe := "recipe1"
	bicepRecipeTemplate := fmt.Sprintf("%s/test/functional/shared/recipes/corerp-redis-recipe:%s", registry, version)
	templateKindBicep := "bicep"

	terraformRecipe := "recipe2"
	terraformRecipeTemplate := "Azure/cosmosdb/azurerm"
	templateKindTerraform := "terraform"
```

File: test/functional/shared/cli/cli_test.go [hunk 4]
```diff
@@ -66,31 +66,37 @@ func verifyRecipeCLI(ctx context.Context, t *testing.T, test shared.RPTest) {
...
 	target := fmt.Sprintf("br:ghcr.io/radius-project/dev/test-bicep-recipes/redis-recipe:%s", generateUniqueTag())
 
 	t.Run("Validate rad recipe register", func(t *testing.T) {
-		output, err := cli.RecipeRegister(ctx, envName, recipeName, templateKind, recipeTemplate, resourceType, false)
-		require.NoError(t, err)
-		require.Contains(t, output, "Successfully linked recipe")
-	})
-
-	t.Run("Validate rad recipe register with insecure registry", func(t *testing.T) {
-		output, err := cli.RecipeRegister(ctx, envName, recipeName, templateKind, recipeTemplate, resourceType, true)
```
[the remark this answers is not in the document]
Comment (sk593) [reply test/functional/shared/cli/cli_test.go]: My bad, missed that this the plainHTTP was set to true here. Reverting to keep this

File: test/functional/shared/cli/cli_test.go (hunk 3 above)
Comment (sk593) [reply test/functional/shared/cli/cli_test.go]: Updated to this

**Review by kachawla — APPROVED** [review]

**Review by lakshmimsft — APPROVED** [review]

**Pull request merged**